### DISCUSSION TITLE
"Shuffle" sort mode

### DISCUSSION THREAD

**Discussion by Kacarott** [codewars/codewars.com#2767] [Ideas]
There are very many nice kata which are largely ignored, due to not being in the top 20 or so kata in any of the sorting criteria. Many of these criteria are fairly static, and are often almost the same as each other (eg. Most completed/positive feedback). Some other criteria change much more frequently (eg. Newest), however still strongly prioritise kata with many translations, due to them being updated often.

My idea is to aid users in search of hidden gems by adding a "shuffle" sorting criteria, which would take all the kata resulting from the other applied filters, and present them in a randomised order.

**Comment by MarcusLassila** [comment]
Excellent idea! Many times have I scrolled through a large number of katas just to find something I have not seen before. A possibility to list them in a randomised order would have been very nice.

**Comment by hobovsky** [comment]
Shuffling does not go too well with paging. Do you know any approach which would allow for paging of randomly ordered items?

**Comment by kazk** [comment]
I'll add `?sample=true` parameter to show random 30 kata matching the search. You can reload the page to get another set.

**Comment by kazk** [comment]
- https://www.codewars.com/kata/?q=&sample=true
- https://www.codewars.com/kata/?q=&beta=true&sample=true
